gnomad_qc/v4/assessment/calculate_per_sample_stats.py

```diff
@@ -54,206 +61,479 @@
+    lof_combos: List[List[str]] = LOF_FILTER_COMBOS,
+    lof_combo_override: Dict[str, List[str]] = LOF_FILTERS_FOR_COMBO,
+    filter_group_key_rename: Dict[str, str] = MAP_FILTER_FIELD_TO_META,
+) -> List[Dict[str, str]]:
+    """
+    Generate list of filter combinations for summary stats.
+
+    This function combines various filter settings for summary statistics and generates
+    all possible filter combinations. It ensures that the generated combinations include
+    both common filters and specific loss-of-function (LOF) filters.
+
+    .. note::
+
+        - The "variant_qc" filter group is removed if the value is "none", which can
+          lead to a filter group of {} (no filters).
+        - The `filter_group_key_rename` parameter can be used to rename keys in the
+          `all_sum_stat_filters`, `common_combo_override`, or `lof_combo_override`
+          after creating all combinations.
+
+    Example:
+        Given the following input:
+
+        .. code-block:: python
+
+            all_sum_stat_filters = {
+                "variant_qc": ["none", "pass"],
+                "capture": ["1", "2"],
+                "max_af": [0.01],
+                "lof_csq": ["stop_gained"],
+                "loftee_flags": ["with_flags"]
+            }
+            common_filter_combos = [["variant_qc"], ["variant_qc", "capture"]]
+            common_combo_override = {"variant_qc": ["pass"], "capture": ["1"]}
+            lof_combos = [ ["lof_csq", "loftee_HC", "loftee_flags"]]
+            lof_combo_override = {"loftee_HC": ["HC"]}
+            filter_group_key_rename = {"lof_csq": "csq","loftee_HC": "loftee_labels"}
+
+        The function will generate the following filter combinations:
+
+        .. code-block:: python
+
+            [
+               # Combinations of all common filter keys and their possible values.
+                {},
+                {'capture': '1'},
+                {'capture': '2'},
+                {'variant_qc': 'pass'},
+                {'variant_qc': 'pass', 'capture': '1'},
+                {'variant_qc': 'pass', 'capture': '2'},
+
+                # Combinations of all requested common filter combinations with all
+                # possible other filter keys and values.
+                {'variant_qc': 'pass', 'max_af': '0.01'},
+                {'variant_qc': 'pass', 'csq': 'stop_gained'},
+                {'variant_qc': 'pass', 'loftee_flags': 'with_flags'},
+                {'variant_qc': 'pass', 'capture': '1', 'max_af': '0.01'},
+                {'variant_qc': 'pass', 'capture': '1', 'csq': 'stop_gained'},
+                {'variant_qc': 'pass', 'capture': '1', 'loftee_flags': 'with_flags'},
+
+                # Combinations of all requested common filter combinations with all
+                # requested LOF filter combination keys and their requested values.
+                {
+                    'variant_qc': 'pass', 'csq': 'stop_gained', 'loftee_labels': 'HC',
```

Comment:
Also we expected to see these:
```commandline
  {'csq_set': 'lof', 'variant_qc': 'pass', 'loftee_labels': 'HC'}
  {'csq_set': 'lof', 'variant_qc': 'pass', 'loftee_labels': 'HC', 'loftee_flags': 'with_flags'}
  {'csq_set': 'lof', 'variant_qc': 'pass', 'loftee_labels': 'HC', 'loftee_flags': 'no_flags'}
    {'csq_set': 'lof', 'variant_qc': 'pass', 'capture': 'ukb_broad_intersect','loftee_labels': 'HC'}
  {'csq_set': 'lof', 'variant_qc': 'pass', 'capture': 'ukb_broad_intersect','loftee_labels': 'HC', 'loftee_flags': 'with_flags'}
  {'csq_set': 'lof', 'variant_qc': 'pass', 'capture': 'ukb_broad_intersect','loftee_labels': 'HC', 'loftee_flags': 'no_flags'}
```
Right?

Reply:
I think I (probably wrongly) assumed that if it had an lof label it was in that consequence set, but I agree that this is more intuitive so I will add